syntax = "proto2";

package vct.col.ast;

import "scalapb/scalapb.proto";

option (scalapb.options) = {
  lenses: false
  getters: false
  no_default_values_in_constructor: true
  package_name: "vct.col.ast.serialize"
};

message Origin {
  repeated OriginContent content = 1;
}

message OriginContent {
  oneof content {
    SourceName source_name = 1;
    PreferredName preferred_name = 2;
    Context context = 3;
    ReadableOrigin readable_origin = 4;
    PositionRange position_range = 5;
    LabelContext label_context = 6;
  }
}

message SourceName {
  required string name = 1;
}

message PreferredName {
  repeated string preferred_name = 1;
}

message Context {
  required string context = 1;
  required string inline_context = 2;
  required string short_position = 3;
}

message ReadableOrigin {
  required string directory = 1;
  required string filename = 2;
  optional string checksum = 3;
  optional string checksum_kind = 4;
}

message PositionRange {
  required sint32 start_line_idx = 1;
  required sint32 end_line_idx = 2;
  optional sint32 start_col_idx = 3;
  optional sint32 end_col_idx = 4;
}

message LabelContext {
  required string label = 1;
}